index.bs

```diff
@@ -6925,19 +6954,51 @@ To <dfn>maybe abort network response body collection</dfn> given |request|:
 
 </div>
 
+<div algorithm>
+To <dfn>maybe collect network request body</dfn> given |request|:
+
+1. Let |collected data| be [=get collected data=] with |request|'s [=request id=] and "request".
+
+1. If |collected data| is null, return.
+
+   NOTE: This might happen if there are no collectors setup when the request is created,
+   and [=clone network request body=] does not clone the corresponding body.
+   Or if the body was null in the first place.
+
+1. [=Maybe collect network data=] with |request|, |collected data|, null and "request".
+
+</div>
+
 <div algorithm>
 To <dfn>maybe collect network response body</dfn> given |request| and |response|:
 
 1. If |response|'s [=response/status=] is a [=redirect status=], return.
 
-   Note: For redirects, only the final response body is stored.
+   NOTE: For redirects, only the final response body is stored.
 
 1. Let |collected data| be [=get collected data=] with |request|'s [=request id=] and "response".
 
 1. If |collected data| is null, return.
 
    NOTE: This might happen if there are no collectors setup when the response is created,
    and [=clone network response body=] does not clone the corresponding body.
+   Or if the body was null in the first place.
+
+1. Let |size| be |response|'s [=response body info=]'s [=encoded size=].
+
+   NOTE: There is a discrepancy between the fact that the bytes retrieved from the
+   fetch stream correspond to the decoded data, but the encoded (network) size is
+   used in order to calculate size limits. Implementations might decide to use a storage
+   model such that it uses less size than storing the decoded data, as long as the data
+   returned to clients in getData is identical to the decoded data. The potential
+   tradeoff between storage and performance is up to the implementation.
+
+1. [=Maybe collect network data=] with |request|, |collected data|, |size| and "response".
+
+</div>
+
+<div algorithm>
+To <dfn>maybe collect network data</dfn> given |request|, |collected data|, |size| and |data type|:
```

Comment:
Even though we usually don't do it, we can provide types here for readability:
```suggestion
To <dfn>maybe collect network data</dfn> given [=/request=] |request|, [=network data=] |collected data|, js-uint |size| and [=network.DataType=] |data type|:
```

Reply:
Done